lib/async/enumerator.rb

```diff
@@ -17,10 +18,31 @@ class Enumerator
 
     # Creates async wrapper for enumerable.
     # @param enumerable [Enumerable] Object to wrap
-    # @param max_fibers [Integer, nil] Concurrency limit
-    def initialize(enumerable, max_fibers: nil)
+    # @param config [Config, nil] Configuration object
+    # @param kwargs [Hash] Configuration options (max_fibers, etc.)
+    def initialize(enumerable = [], config = nil, **kwargs)
       @enumerable = enumerable
-      @max_fibers = max_fibers
+
+      # Handle backward compatibility - if config looks like old max_fibers keyword arg
+      if config.is_a?(Hash) && config.key?(:max_fibers) && kwargs.empty?
+        kwargs = config
+        config = nil
+      end
```

Comment:
this isn't just for backwards compatibility, we want `Enumerator.new(max_fibers: ...)` to be the standard way to set max_fibers for users of this class. the config object is something we use internally and for more advanced use cases.